go/runtime/registry/config.go

```diff
@@ -13,11 +13,16 @@ import (
 	"github.com/oasisprotocol/oasis-core/go/config"
 	cmdFlags "github.com/oasisprotocol/oasis-core/go/oasis-node/cmd/common/flags"
 	"github.com/oasisprotocol/oasis-core/go/runtime/bundle"
+	"github.com/oasisprotocol/oasis-core/go/runtime/bundle/component"
 	"github.com/oasisprotocol/oasis-core/go/runtime/history"
 )
 
-func getLocalConfig(runtimeID common.Namespace) map[string]any {
-	return config.GlobalConfig.Runtime.GetLocalConfig(runtimeID)
+func getLocalConfig(runtimeID common.Namespace, compID component.ID) map[string]any {
+	compCfg, ok := config.GlobalConfig.Runtime.GetComponent(runtimeID, compID)
+	if !ok || len(compCfg.Config) == 0 {
```

Comment:
This is strange. If component doesn't have local config, the global one is used. But if it has one, the global one is ignored and not overwritten. Should it be?

Reply:
> is ignored and not overwritten

What do you mean, it is overwritten? Do you mean that we would merge configurations? The global configuration is usually meant for the RONL component and makes no sense in other components. Eventually we should probably get rid of global configuration and only have per-component one.

Reply:
Should in that case this be
```go
	// Support legacy RONL configuration defined at the top level.
	if !ok && compID == component.ID_RONL {
```
and runtime config updated to
```go
// RuntimeConfig is the runtime configuration.
type RuntimeConfig struct {
	// Config contains runtime local configuration.
	// NOTE: This may go away in the future, use `Component.Config` instead.
	Config map[string]any `yaml:"config,omitempty"`
```

Reply:
Fixed.